testsuite/policy/authorization/sections.py

```diff
@@ -67,10 +67,18 @@ def section(self):
         """The actual dict section which will be edited"""
         return self.obj.auth_section.setdefault(self.section_name, {})
 
-    def add_item(self, name: str, value: dict, **common_features):
+    def add_item(self, name: str, value: dict, *, defaults: bool = False, overrides: bool = False, **common_features):
```

Comment:
How will this work with AuthConfig?

Reply:
AuthConfig doesn't support D&O. Only AuthPolicy and RateLimitPolicy

Reply:
Thats precisely why I am asking, Sections are used by both AuthConfig and AuthPolicy

Reply:
The idea was "just don't use it with AuthConfig" 😄 . Pytest marks should restrict this. Do you want this to be completely forbidden?

Reply:
Well, I think we are solving defaults in a wrong place. My issues with this solution:
1) It doesnt clearly show how defaults are applied:
  * Consider how we add stuff now: `authorization.identity.add_oidc` -> this clearly indicates that we go into identity section and add some infor about OIDC. It could be better but I still think it conveys information nicely. 
  * Now look how default are done `authorization.identity.add_oidc(default=True)` ->If I read this I will assume it is the same section but we add default parameter somewhere (also could be only me), but that is not true. We need to add in a completely different section (`defaults` instead of `rules`) and I would never guessed it in a first place nor would I be able to replicate it directly in AP.
2) It is used by AuthConfig as well. While this could be easily ignored, it adds another bit of abstraction magic to already magic class and now will not work universally, as was originally intended, but that was long time ago and we can agree to change it.

My preferred way to deal with D&O would be like this as you are adding it to a completely different section:
`authorization.defaults.identity.add_oidc`, `authorization.overrides.identity.add_oidc`, `authorization.rules.identity.add_oidc` (this could be default if you dont specify anything to be backwards compatible).

This would be trickier to implement but I think it would be a cleaner solution. Question is if this solution (or any other cleaner one) warrants the time investment needed.

Reply:
These examples are definitely cleaner. I'll look what I can do. Do you want to merge these tests in the meantime? I think it is easily refactorable with what I have now.